lib/src/chunk_builder.dart

```diff
@@ -333,6 +341,48 @@ class ChunkBuilder {
     preserveNewlines(linesBeforeToken);
   }
 
+  /// Writes the text of [comment].
+  ///
+  /// If it's a JavaDoc comment that should be fixed to use `///`, fixes it.
+  void _writeCommentText(SourceComment comment) {
+    if (_formatter.fixes.contains(StyleFix.docComments)) {
+      var match = _javaDocComment.firstMatch(comment.text);
+      if (match != null) {
```

Comment:
[optional] invert the condition and short circuit return would reduce nesting in the rest of the method.

Same with the condition on line 348.